src/host/posix/cli_daemon.cpp

```diff
@@ -71,6 +71,39 @@ std::string CliDaemon::GetSocketFilename(const std::string &aNetIfName, const ch
     return fileName;
 }
 
+void CliDaemon::HandleCommandOutput(const char *aOutput)
+{
+    int    ret;
+    size_t length = strlen(aOutput);
+
+    VerifyOrExit(mSessionSocket != -1);
+
+    static_assert(sizeof(kTruncatedMsg) < kCliMaxLineLength, "OTBR_CONFIG_CLI_MAX_LINE_LENGTH is too short!");
+
+    if (length >= kCliMaxLineLength)
+    {
+        length = kCliMaxLineLength - 1;
+        memcpy(const_cast<char *>(aOutput) + kCliMaxLineLength - sizeof(kTruncatedMsg), kTruncatedMsg,
```

Comment:
I think we are assuming that since we get `aOutput` from the NCP spinel decoder (which points to a buffer we own in another module), we can cast and remove `const` from `aOutput` and overwrite its content here? This should work fine for now, but this could cause issues if things change and `aOutput` should be treated as a true `const char *` (i.e., its content should not be changed).

Perhaps we should copy the content into a new string to make the code safer. I believe we can even use `std::string` here within `ot-br-posix`.